### PR TITLE
[MAINTENANCE] Plot-level dropdown for `DataAssistantResult` display charts

### PR DESCRIPTION
**Pull request by NathanFarmer**
Changes proposed in this pull request:
- GREAT-1020
- Add a dropdown that wraps all `DataAssistantResult` metrics or expectations + metrics charts so that the charts don't flood the notebook

Previous Design Review notes:
- Displaying all available charts is crowding the notebook

### Definition of Done
- [X] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [X] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [X] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: requirements-dev-lite.txt [hunk 1]
```diff
@@ -1,7 +1,6 @@
 boto3==1.17.106 # This should match the version in constraints-dev.txt
 flask>=1.0.0 # for s3 test only (with moto)
 freezegun>=0.3.15
-ipywidgets>=7.5.1
```
Comment (NathanFarmer) [inline requirements-dev-lite.txt:4]: Tests were failing because this library wasn't in `requirements.txt` so I removed it from here so it wasn't duplicated.

File: great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py [hunk 2]
```diff
@@ -541,31 +544,118 @@ def _display(
...
+            options=chart_titles,
+            description="Select Plot: ",
+            style={"description_width": "initial"},
+            layout={"width": "max-content", "margin": "0px"},
+        )
+
+        # As of 19 July, 2022 there is a DeprecationWarning due to the latest ipywidgets' interaction with
+        # ipykernel versions > 6.0.0. Rather than add a version constraint to ipykernel, we suppress
+        # DeprecationWarnings produced by module ipywidgets.widgets.widget_output
+        warnings.filterwarnings(
```
Comment (Kilo59) [inline great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py]: Do we have a backlog item to keep track of this?
Do you know if there's a timeline for when this deprecated feature might be removed?

Comment (NathanFarmer) [reply great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py]: There is nothing else we can do (that I know of) to suppress this warning, other than constrain the version of `ipykernel` - this is an issue with `ipywidgets` unfortunately. I don't see any issues in their list of [570 GitHub issues](https://github.com/jupyter-widgets/ipywidgets/issues) that discuss this warning, so I'm not sure what the timeline looks like.

> DeprecationWarning: Kernel._parent_header is deprecated in ipykernel 6. Use .get_parent()

File: great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py [hunk 3]
```diff
@@ -541,31 +544,118 @@ def _display(
...
+                        break
+                    except AttributeError:
+                        continue
+
+            if chart_title is not None:
+                chart_titles.append(chart_title)
+            else:
+                raise ge_exceptions.DataAssistantResultExecutionError(
+                    "All DataAssistantResult charts must have a title."
+                )
```
Comment (Kilo59) [inline great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py]:
```suggestion
            if not chart_titles:
                raise ge_exceptions.DataAssistantResultExecutionError(
                    "All DataAssistantResult charts must have a title."
                )
           chart_titles.append(chart_title)
```

Using a guard clause here might make this slightly easier to grok. Assuming I've understood the issue correctly 😄 .
https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88

Apologies the above article is for Golang but hopefully the examples, explanations are easy to follow.

There's actually a python-bytes podcast episode that mentions this.
https://pythonbytes.fm/episodes/show/131/python-3-has-issues-over-on-github

File: great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py (hunk 2 above)
Comment (Kilo59) [reply great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py]: I meant should we create a backlog item to implement the change suggested by the deprecation warning?

If we don't want to (or can't) use `.get_parent()` right now, we should probably track this otherwise we are likely to forget about it and have it just break on us one-day when the library maintainers remove this piece of code.

Comment (NathanFarmer) [reply great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py]: @Kilo59 there is no change that can be made on our side. The change needs to be made by the developers of `ipywidgets`.

Comment (Kilo59) [reply great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py]: Oh lol, sorry, I just glazed over the difference between `ipywidgets` vs `ipykernel`

File: tests/test_deprecation.py [hunk 4]
```diff
@@ -40,7 +40,7 @@ def test_deprecation_warnings_are_accompanied_by_appropriate_comment(
             contents = f.read()
 
         matches: List[str] = regex_for_deprecation_comments.findall(contents)
-        warning_count: int = contents.count("DeprecationWarning")
+        warning_count: int = contents.count("DeprecationWarning(")
```
Comment (Kilo59) [inline tests/test_deprecation.py]: Why this change to add the opening param?

**Review by Kilo59 — APPROVED** [review]

File: great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py (hunk 3 above)
Comment (NathanFarmer) [reply great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py]: I came up with something even simpler:

```
            if chart_title is None:
                raise ge_exceptions.DataAssistantResultExecutionError(
                    "All DataAssistantResult charts must have a title."
                )

            chart_titles.append(chart_title)
```

File: tests/test_deprecation.py (hunk 4 above)
Comment (NathanFarmer) [reply tests/test_deprecation.py]: This ended up being a mistake.
